### PR TITLE
Fix APPLICATION_CANCELLED by handling back button press

### PR DESCRIPTION
**Pull request by nickbopp**
`AuthorizationActivity` no longer needs to directly handle back button presses, and it's all standard machinery.  `(Web)AuthorizationFragment` handle the press, and lifetime is tied to that of its view.

This change also updates the required androidx appcompat version from 1.0.2 -> [1.1](https://developer.android.com/jetpack/androidx/releases/appcompat#version_110_3).  I expect most end-user applications are actually using an even newer version of this package already, as even 1.1 was released in 2019.

### CODE REVIEW

**Comment by nickbopp** [comment]
I'm still working on getting a good MSAL and OneAuth test pass using these Common changes, but I wanted to give folks a chance to look at the code.

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java [hunk 1]
```diff
@@ -198,14 +198,10 @@
...
 
-    /**
-     * NOTE: Fragment-only mode will not support this, as we don't own the activity.
-     * This must be invoked by AuthorizationActivity.onBackPressed().
-     */
     @Override
-    public boolean onBackPressed() {
+    public void handleBackButtonPressed() {
         final String methodTag = TAG + ":onBackPressed";
-        Logger.info(methodTag, "Back button is pressed");
```
Comment (iamgusain) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java:210]: nit: update the methodTag to include the new name for the method

**Comment by iamgusain** [comment]
nit: Can you also add a link to any documentations on what changed between appCompat version 1.1.0 and 1.0.2 corresponding to the change in this PR, for future reference.

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java [hunk 2]
```diff
@@ -51,11 +51,4 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         }
         setFragment(mFragment);
     }
-
```
Comment (iamgusain) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java:54]: do we also need to remove this in "CurrentTaskAuthorizationActivity,java"

**Review by iamgusain — COMMENTED** [review]
added few minor comments

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java (hunk 2 above)
Comment (nickbopp) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java:54]: Yes, thanks for the catch.

**Comment by nickbopp** [comment]
When this PR is ingested to MSAL/OneAuth, the `addToBackStack` calls in the two EmbeddedBrowser classes should be removed.  These changes together solve APPLICATION_CANCELLED + fragment lifetime being too long.

**Review by irving-as — APPROVED** [review]

**Review by iamgusain — APPROVED** [review]

**Review by shahzaibj — APPROVED** [review]

**Comment by nickbopp** [comment]
Okay, I'm back from OOF so this is ready for merge now, when someone with permissions is available to do so.

**Pull request merged**